base/show.jl

```diff
@@ -241,6 +241,14 @@ end
 
 show(io::IO, x::DataType) = show_datatype(io, x)
 
+# Check whether 'sym' (defined in module 'parent') is reachable from module 'from'
+# If an object with this name exists in 'from', we need to check it's the same object
+function is_reachable(sym::Symbol, parent::Module, from::Module)
+    isdefined(from, sym) || return false
+    getfield(from, sym) === getfield(parent, sym)
+end
+is_reachable(sym::Symbol, parent::Module, from::Void) = false
```

Comment:
as `isdefined(nothing, :anysymbol) == false`, the 2 methods could be simplified to `is_reachable(sym::Symbol, parent::Module, from::Union{Module,Void}) = isdefined(from, sym) && getfield(from, sym) === getfield(parent, sym)`. Not saying it's necessarily an improvement!

Reply:
Yeah, at least the current definition is quite clear, if a bit verbose.

Reply:
I find @rfourquet's formulation clearer, but people's brains are different, so YMMV. This isn't a public function so the name is not crucial, but this might be better called `isvisible`; `isreachable` sounds like there's some kind of recursive search going on, which there isn't – it's just an immediate visibility check.